galasa-managers-parent/galasa-managers-visualisation-parent/dev.galasa.elasticlog.manager/src/main/java/dev/galasa/elasticlog/internal/ElasticLogManagerImpl.java

```diff
@@ -0,0 +1,227 @@
+	    String request = this.gson.toJson(this.runProperties);
+	    logger.info("Sending Run Request to ElasticLog Endpoint");
+	    logger.trace("Document Request -\n" + request);
+	    sendRunData(request);
+    }
+    
+    /**
+     * Create the required indexes and send document requests
+     * 
+     * @param request - document request json
+     * @throws ElasticLogManagerException
+     */
+    private void sendRunData(String request) throws ElasticLogManagerException {
+        //Register endpoint data as confidential
+    	String index = ElasticLogIndex.get();
+    	String endpoint = ElasticLogEndpoint.get();
+		ctf.registerText(index, "ElasticLog Index");
+		ctf.registerText(endpoint, "ElasticLog Endpoint");
+    	try {
+            //Create mapping Json
+    		String mapping = createIndexMapping();
+    		logger.trace("Index Mapping Request -\n" + mapping);
+            
+            //Set up http client for requests
+            client.setTrustingSSLContext();
+            client.addOkResponseCode(201);
+            client.setURI(new URI(endpoint));
+            
+            //Create index if not present
+    		HttpClientResponse<String> response = client.head(endpoint + "/" + index);
+    		if (response.getStatusCode() == 404) 
+                client.putJson(index, mapping, false);
+            
+            //Send document to index
+			client.postJson(index + "/_doc", request, false);
+        
+            //Create latest index if not present
+    		index = index + "_latest";
+    		HttpClientResponse<String> indexResponse = client.head(ElasticLogEndpoint.get() + "/" + index);
+			if (indexResponse.getStatusCode() == 404)
+                client.putJson(index, mapping, false);
+        
+            //Obtain id of document if test case is already recorded
+        	String id = null;
+        	String testCase = (String) this.runProperties.get("testCase");
+			String searchResponse = client.get(index + "/_search?q=testCase:" + testCase);
+			if (searchResponse.contains(testCase)) {
+				JsonObject json = this.gson.fromJson(searchResponse, JsonObject.class);
+				id = json.get("hits").getAsJsonObject()
+                         .get("hits").getAsJsonArray()
+                         .get(0).getAsJsonObject()
+                         .get("_id").getAsString();
+			}
+            
+            //Delete document if already present then send new document
+    		if (id != null)
+    			client.delete(index + "/_doc/" + id, false);
+    		client.postJson(index + "/_doc", request, false);
+
+		} catch (HttpClientException e) {
+			throw new ElasticLogManagerException("Error in sending request to ElasticLog Endpoint", e);
+		} catch (URISyntaxException e) {
+			throw new ElasticLogManagerException("Error in parsing ElasticLog Endpoint to URI", e);
+		}
+    }
+
+    /**
+     * Create a mapping properties request from run properties
+     * 
+     * @return - Mapping properties request
+     */
+    private String createIndexMapping() {
```

Comment:
Why cant Elastic do this itself.   I can't believe that all producers of information for elastic have to go through this process.